lib/afterpay/http_service/request.rb

```diff
@@ -33,7 +33,13 @@ def perform(params = {})
       # Creates memoized Faraday::Connection instance
       # @return [Faraday::Connection] Faraday::Connection instance
       def adapter
-        @adapter ||= Faraday.new(server, {}, &middleware)
+        @adapter ||= Faraday.new(server, { "headers" => { "User-Agent" => option_user_agent } }, &middleware)
+      end
+
+      def option_user_agent
+        return {} unless ::Afterpay.user_agent
```

Comment:
This won't work because the user_agent should be a string while you are returning an hash.

I suggest you to rename this method `faraday_options` and to return the whole `{ headers: { "User-Agent" => ::Afterpay.user_agent } }` if the user_agent is not nil, otherwise return an empty hash.

Reply:
Done!